llvm/lib/Transforms/Scalar/StructurizeCFG.cpp

```diff
@@ -990,10 +996,10 @@ BasicBlock *StructurizeCFG::getNextFlow(BasicBlock *Dominator) {
                                         Func, Insert);
   FlowSet.insert(Flow);
 
-  // use a temporary variable to avoid a use-after-free if the map's storage is
-  // reallocated
-  DebugLoc DL = TermDL[Dominator];
-  TermDL[Flow] = std::move(DL);
+  auto *Term = Dominator->getTerminator();
+  if (const DebugLoc &DL =
```

Comment:
We used to do a copy of the `DebugLoc` because the reference to `TermDL[Dominator]` could be invalidated if the underlying container reallocates a bigger array when we do `TermDL[Flow]`.

Could this issue still happen here? If yes, we would need to do a copy of the `DebugLoc`.

Reply:
Thinking about this more, expression:

Term ? Term->getDebugLoc() : TermDL.lookup(Dominator)

returns temporary DebugLoc object because this is the common type for LHS and RHS of the operator? because LHS returns reference and RHS - temporary object. This can work as is without problems, but I don't like that temporary DL would require redundant "retrack" of the internal MDNode pointer and I better avoid it by rewriting the statement.

Reply:
Done.